healthcheck.go

```diff
@@ -13,33 +17,94 @@ type healthService struct {
 }
 
 type healthConfig struct {
-	appSystemCode string
-	appName       string
-	port          string
+	appDesc                  string
+	appSystemCode            string
+	appName                  string
+	port                     string
+	writerHealthURI          string
+	contentResolverHealthURI string
+	producer                 producer.MessageProducer
+	client                   *http.Client
 }
 
 func newHealthService(config *healthConfig) *healthService {
 	service := &healthService{config: config}
 	service.checks = []health.Check{
-		service.sampleCheck(),
+		service.writerCheck(),
+		service.contentResolverCheck(),
+		service.producerCheck(),
 	}
 	return service
```

Comment:
If the return statement is a pointer, then for readability purposes it is better to have the address reference here (&service), and at the var definition have it as a simple struct.